File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        if(nums.length == 0) return -1;
        int maj = nums[0];
        int count = 1;

        for(int i = 1; i < nums.length; i++){
            if (maj == nums[i]){
                count++;
            }else{
                count--;
            }

            if (count == 0){
                maj = nums[i];
                count = 1;
            }
        }

        return maj;
        
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if not nums:
            return -1
        count = 1
        n = nums[0]

        for i in range(1,len(nums)):
            if nums[i] == n:
                count += 1
            else:
                count -= 1
            if count == 0:
                n = nums[i]
                count = 1
        
        return n

        